packages/edit-post/src/components/header/header-toolbar/index.js

```diff
@@ -55,7 +68,9 @@ function HeaderToolbar() {
 
 	return (
 		<NavigableToolbar
-			className="edit-post-header-toolbar"
+			className={ classnames( 'edit-post-header-toolbar', {
+				'show-icon-labels': showIconLabels && isMediumOrLarger,
+			} ) }
```

Comment:
in this case, it is difficult to pass `showIconLabels` prop directly to buttons, this is why I went with pseudo-elements here